### ISSUE TITLE
[SeparatorList] Add SeparatorList component

### ISSUE THREAD

**Issue by akdetrick** [narmi/design_system#343]
### Short Description

A layout component that adds gutters and dividers between items. The dividers should be rendered as CSS pseudo elements and styled so that they are hidden from screen readers.

This component is useful for breadcrumbs, pipe lists, and middot lists.

### Mockup
<img width="282" alt="Screen Shot 2021-11-04 at 4 50 06 PM" src="https://user-images.githubusercontent.com/231252/140418164-1d006f8f-fce1-4b3d-bb20-d46b01413ff8.png">

### Responsibilities

**This component should be responsible for the following:**
- render separators between items
- color configuration of separators
- rendering each item as inline-flex
- placing gutters between items and separators

**This component should NOT be responsible for...**
- individual separator configuration (they all need to be the same character and same color)

### Props Signature

```
<SeparatorList
  items={[ JSXNode, JSXNode, JSXNode ]}
  separator="|"
  separatorColor={NDS_TEXT_HINT}
/>
```

```
SeparatorList.propTypes = {
  items: arrayOf nodes,
  separator: string, // the character to render in pseudo elements
  separatorColor: string
}
```
